Algorithm.CSharp/CompositeRiskManagementModelFrameworkAlgorithm.cs

```diff
@@ -0,0 +1,103 @@
+/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System.Collections.Generic;
+using QuantConnect.Algorithm.Framework;
+using QuantConnect.Algorithm.Framework.Alphas;
+using QuantConnect.Algorithm.Framework.Execution;
+using QuantConnect.Algorithm.Framework.Portfolio;
+using QuantConnect.Algorithm.Framework.Risk;
+using QuantConnect.Algorithm.Framework.Selection;
+using QuantConnect.Interfaces;
+
+namespace QuantConnect.Algorithm.CSharp
+{
+    /// <summary>
+    /// Show cases how to use the <see cref="CompositeRiskManagementModel"/> to define
+    /// </summary>
+    public class CompositeRiskManagementModelFrameworkAlgorithm : QCAlgorithmFramework, IRegressionAlgorithmDefinition
+    {
+        public override void Initialize()
+        {
+            // Set requested data resolution
+            UniverseSettings.Resolution = Resolution.Minute;
+
+            SetStartDate(2013, 10, 07);  //Set Start Date
+            SetEndDate(2013, 10, 11);    //Set End Date
+            SetCash(100000);             //Set Strategy Cash
+
+            // set algorithm framework models
+            SetUniverseSelection(new ManualUniverseSelectionModel(QuantConnect.Symbol.Create("SPY", SecurityType.Equity, Market.USA)));
+            SetAlpha(new ConstantAlphaModel(InsightType.Price, InsightDirection.Up, System.TimeSpan.FromMinutes(20), 0.025, null));
+            SetPortfolioConstruction(new EqualWeightingPortfolioConstructionModel());
+            SetExecution(new ImmediateExecutionModel());
+
+            // define risk management model as a composite of several risk management models
+            SetRiskManagement(new CompositeRiskManagementModel(
+                new NullRiskManagementModel(),
```

Comment:
For the purposes of this regression algorithm, I think we should have 2 valid risk management models here. I've modified the `MaximumDrawdownPercentPerSecurity` risk model to be inverted (take profit). You can copy this class into this algorithm file and use it as your second risk model. Be sure both risk models perform adjustments so we know both are being executed.
```csharp
/// <summary>
/// Provides an implementation of <see cref="IRiskManagementModel"/> that limits the unrealized profit
/// per holding to the specified percentage.
/// </summary>
public class MaximumUnrealizedProfitPercentPerSecurity : RiskManagementModel
{
    private readonly decimal _unrealizedProfitPercent;

    /// <summary>
    /// Initializes a new instance of the <see cref="MaximumDrawdownPercentPerSecurity"/> class
    /// </summary>
    /// <param name="unrealizedProfitPercent">The maximum percentage unrealized profit allowed for any single security holding, defaults to 5% unrealized profit per security</param>
    public MaximumUnrealizedProfitPercentPerSecurity(
        decimal unrealizedProfitPercent = 0.05m
        )
    {
        _unrealizedProfitPercent = Math.Abs(unrealizedProfitPercent);
    }

    /// <summary>
    /// Manages the algorithm's risk at each time step
    /// </summary>
    /// <param name="algorithm">The algorithm instance</param>
    /// <param name="targets">The current portfolio targets to be assessed for risk</param>
    public override IEnumerable<IPortfolioTarget> ManageRisk(QCAlgorithmFramework algorithm, IPortfolioTarget[] targets)
    {
        foreach (var kvp in algorithm.Securities)
        {
            var security = kvp.Value;

            if (!security.Invested)
            {
                continue;
            }

            var pnl = security.Holdings.UnrealizedProfitPercent;
            if (pnl > _unrealizedProfitPercent)
            {
                // liquidate/take profit
                yield return new PortfolioTarget(security.Symbol, 0);
            }
        }
    }
}
```

Reply:
Or feel free to include your own custom risk model! The more the merrier :)  Just include it as part of the algorithm (nested class).